app/[locale]/academics/helper/courses/AddCourseModal.tsx

```diff
@@ -0,0 +1,245 @@
+import { postCourseAction } from '@/actions/academics';
+
+import Dropdown from '@/components/common/form/Dropdown';
+import BasicTextInput from '@/components/editor/common/BasicTextInput';
+import Fieldset from '@/components/editor/common/Fieldset';
+import ModalFrame from '@/components/modal/ModalFrame';
+
+import { Classification, CLASSIFICATION, Course, GRADE, StudentType } from '@/types/academics';
+import { getKeys } from '@/types/object';
+
+import { validateCourseForm } from '@/utils/formValidation';
+import { handleServerAction } from '@/utils/serverActionError';
+import { errorToast, successToast } from '@/utils/toast';
+
+import useCourseEditor from './useCourseEditor';
+
+const getInitCourse = (type: StudentType): Course => ({
+  code: '',
+  credit: 3,
+  grade: type === 'graduate' ? 0 : 1,
+  studentType: type,
+  ko: { name: '', description: '', classification: '전공필수' },
+  en: { name: '', description: '', classification: 'RM' },
+});
+
+export default function AddCourseModal({
+  onClose,
+  studentType,
+}: {
+  onClose: () => void;
+  studentType: StudentType;
+}) {
+  const { content, setContentByKey, setLanguageContent, setClassification } = useCourseEditor(
+    getInitCourse(studentType),
+  );
+
+  const handleSubmit = async () => {
+    try {
+      validateCourseForm(content);
+      handleServerAction(await postCourseAction(content));
+      successToast('새 교과목을 추가했습니다.');
+      onClose();
+    } catch (e) {
+      if (e instanceof Error) {
+        errorToast(e.message);
+      } else {
+        throw e;
+      }
```

Comment:
else문 실행되는 경우는 사실상 없을 것 같은데 utils/error.ts에 `errorToStr`쓰는건 어때?

```suggestion
       errorToast(errorToStr(e));
```

Reply:
저런 유틸이 있었다니